Aspire 9 support

aaronpowell opened the pull request:
<!-- 🚨 Please Do Not skip any instructions and information mentioned below as they are all required and essential to evaluate and test the PR. By fulfilling all the required information you will be able to reduce the volume of questions and most likely help merge the PR faster 🚨 -->

<!-- ⚠️ We will not merge the PR to the main repo if your changes are not in a *feature branch* of your forked repository. Create a new branch in your repo, **do not use the `main` branch in your repo**! -->

<!-- ⚠️ **Every PR** needs to have a linked issue and have previously been approved. PRs that don't follow this will be rejected. >

<!-- 👉 It is imperative to resolve ONE ISSUE PER PR and avoid making multiple changes unless the changes interrelate with each other -->

<!-- 📝 Please always keep the "☑️ Allow edits by maintainers" button checked in the Pull Request Template as it increases collaboration with the Toolkit maintainers by permitting commits to your PR branch (only) created from your fork. This can let us quickly make fixes for minor typos or forgotten StyleCop issues during review without needing to wait on you doing extra work. Let us help you help us! 🎉 -->

Closes #56

<!-- Add an overview of the changes here -->
This pull request includes several updates and improvements across multiple files, primarily focusing on upgrading the Aspire SDK version, removing unnecessary workload installations, and adding new NuGet configurations. Below is a summary of the most important changes:

### Workflow Updates:
* Removed the installation of the Aspire workload in the `dotnet-ci.yml`, `dotnet-main.yml`, and `dotnet-release.yml` files. (`.github/workflows/dotnet-ci.yml` [[1]](diffhunk://#diff-bd6c1bcfcaa13ae3662b9fbb7e13fa55a46715d4ba656f6d663717d309101b6dL64-L65) `.github/workflows/dotnet-main.yml` [[2]](diffhunk://#diff-21899fb7aeae0b752bd2399dabcffa6ded4374d82e1ccc1e9ecd90b319d832fdL64-L65) `.github/workflows/dotnet-release.yml` [[3]](diffhunk://#diff-e6db66c698d5cacbdcef2155b9658f2d7d603f53c2bd5b75217af9b408247962L68-L69)

### SDK Version Upgrades:
* Updated the `AspireVersion` from `8.2.0` to `9.0.0-preview.4.24504.1` in `Directory.Build.props`. (`Directory.Build.props` [Directory.Build.propsL10-R10](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eL10-R10))
* Added the `Aspire.AppHost.Sdk` with version `9.0.0-preview.4.24477.2` to various project files. (`examples/java/Aspire.CommunityToolkit.Hosting.Java.AppHost/Aspire.CommunityToolkit.Hosting.Java.AppHost.csproj` [[1]](diffhunk://#diff-65419221cdbebbe325ad15c146c4c95f7778fa1a9b4adc59fe5a411d6b887f50R2) `examples/nodejs-ext/Aspire.CommunityToolkit.Hosting.NodeJS.Extensions.AppHost/Aspire.CommunityToolkit.Hosting.NodeJS.Extensions.AppHost.csproj` [[2]](diffhunk://#diff-775122f6ec789a3032e40f20ef604fe286c364383a5704bd40081e7570642e83R2) `examples/ollama/Aspire.CommunityToolkit.Hosting.Ollama.AppHost/Aspire.CommunityToolkit.Hosting.Ollama.AppHost.csproj` [[3]](diffhunk://#diff-d71de5a842868e4a424210dd22e7d2aa6df01ef0346dce87757875d9eb20d9faR2) `examples/swa/Aspire.CommunityToolkit.StaticWebApps.AppHost/Aspire.CommunityToolkit.StaticWebApps.AppHost.csproj` [[4]](diffhunk://#diff-531c593adada0d20e2dd87ebace5629c973e19f7626e1ee4eb70031b05e5f9a0R2)

### Configuration Additions:
* Added a new `nuget.config` file with specific package sources and mappings for Aspire packages. (`nuget.config` [nuget.configR1-R15](diffhunk://#diff-dbd515845a90a91f83cd2dbcdf41150bff6a75806900384a44afb1f0650d9cc9R1-R15))

### Test File Updates:
* Added `using Aspire.Hosting;` to several test files to ensure proper namespace usage. (`tests/Aspire.CommunityToolkit.Hosting.Java.Tests/ContainerResourceCreationTests.cs` [[1]](diffhunk://#diff-72e05da8ed033186b72f050ac1b2f5559756a4c2592bcb1b929576feacda23fdR1-R2) `tests/Aspire.CommunityToolkit.Hosting.Java.Tests/ExecutableResourceCreationTests.cs` [[2]](diffhunk://#diff-8bdc0074ba1c337eba402b09d81fac8e727e84cbf9a5fc63f0c86186e66ca292R1-R2) `tests/Aspire.CommunityToolkit.Hosting.NodeJS.Extensions.Tests/ResourceCreationTests.cs` [[3]](diffhunk://#diff-e8f9b580041678156de63402b7fada8fe4c8738947726148474dabd5ca538ea9R1)
<!-- All details should be in the linked issue. Feel free to call out any outstanding differences here. -->

## PR Checklist

<!-- Please check if your PR fulfills the following requirements, and remove the ones that are not applicable to the current PR -->

- [ ] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [ ] Based off latest main branch of toolkit
- [ ] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [ ] New integration
  - [ ] Docs are written
  - [ ] Added description of major feature to project description for NuGet package (4000 total character limit, so don't push entire description over that)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Every new API (including internal ones) has full XML docs
- [ ] Code follows all style conventions

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
Please note that breaking changes are likely to be rejected within minor release cycles or held until major versions. -->

## Other information

<!-- Please add any other information that might be helpful to reviewers. -->

Alirexaa commented:
@aaronpowell, We removed the `AddOllama` deprecated API in https://github.com/CommunityToolkit/Aspire/pull/129 and removed the associated API in the unshipped file but we should added the `*REMOVED*` prefix instead of removing it from file.
(the correct way for removing shipped API from these file is to have the API in both files but the unshipped one should have a `*REMOVED*` prefix)
The next time you run `mark-shipped.cmd` (after a new release), these APIs will be removed from the file.
This issue occurred because we removed deprecated API before running `mark-shipped.cmd`.
I Fixed in https://github.com/CommunityToolkit/Aspire/pull/87/commits/cfaf3ab2176e5a144b52f5fa7bce77e9d0762ad6 and https://github.com/CommunityToolkit/Aspire/pull/87/commits/b41447ff87e37e7315f959e52561ada9c925a1ca

aaronpowell commented:
> @aaronpowell, We removed the `AddOllama` deprecated API in #129 and removed the associated API in the unshipped file but we should added the `*REMOVED*` prefix instead of removing it from file. (the correct way for removing shipped API from these file is to have the API in both files but the unshipped one should have a `*REMOVED*` prefix) The next time you run `mark-shipped.cmd` (after a new release), these APIs will be removed from the file. This issue occurred because we removed deprecated API before running `mark-shipped.cmd`. I Fixed in [cfaf3ab](https://github.com/CommunityToolkit/Aspire/commit/cfaf3ab2176e5a144b52f5fa7bce77e9d0762ad6) and [b41447f](https://github.com/CommunityToolkit/Aspire/commit/b41447ff87e37e7315f959e52561ada9c925a1ca)

Thanks - still getting the hang of this API analyzer

aaronpowell commented:
With only ~2 weeks until Aspire 9 is released, we're probably getting close to when it'd be worthwhile merging this into the `main` branch and cutting `main` over to being Aspire 9 exclusively.

How about next Monday for the cutoff date for anything under an 8.x release (unless a super critical bug comes up)?
